crypto/ct/ct_b64.c

```diff
@@ -45,9 +45,11 @@ static int ct_base64_decode(const char *in, unsigned char **out)
         goto err;
     }
 
-    /* Subtract padding bytes from |outlen| */
-    while (in[--inlen] == '=') {
+    /* Subtract padding bytes from |outlen|.  Any more than 2 is malformed. */
+    i = 0;
+    while (in[--inlen] == '=' && i < 2) {
         --outlen;
+        ++i;
     }
```

Comment:
So, errr, if the input turns out to be malformed, shouldn't this cause an error?  'cause the way it looks right now, sure, you stop the downcount of `outlen`, but you still return `outbuf` like there's nothing wrong with it?  That doesn't quite add up...